### ISSUE TITLE
Sui 101, Access On-Chain Time - test fails 

### ISSUE THREAD

**Issue by rwstapleton0** [MystenLabs/sui#15226]
As title suggests the test on the 'Sui 101, Access On-Chain Time' fails. This is due to the create_for_testing returning a clock object that is not handled, doc suggests that this helper function did share the object too, instead they are now separate functions. 

solutions are: 
destroy this object with 'clock::destroy_for_testing(clock)'
```
let ts = test::begin(@0x0);
let clock = clock::create_for_testing(ts::ctx(&mut ts));

clock::increment_for_testing(&mut clock, 20);
clock::increment_for_testing(&mut clock, 22);
assert!(clock::timestamp_ms(&clock) == 42, 0);

clock::destroy_for_testing(clock);
ts::end(ts);
```

share the object with 'clock::share_for_testing(clock)' helper function. requires 2 transactions.
```
let ts = ts::begin(@0x0);
{
    ts::next_tx(&mut ts, ADMIN);
    let clock = clock::create_for_testing(ts::ctx(&mut ts));
    clock::share_for_testing(clock);
};
{
    test::next_tx(&mut ts, ADMIN);
    let clock = test::take_shared<Clock>(&ts);

    clock::increment_for_testing(&mut clock, 20);
    clock::increment_for_testing(&mut clock, 22);
    assert!(clock::timestamp_ms(&clock) == 42, 0);
    
    ts::return_shared(clock);
};
ts::end(ts);
```
new to sui and move, may have missed something or there is a more concise way of writing this :)

**Comment by stefan-mysten** [comment]
@rwstapleton0 thanks for reporting this issue!

**Comment by amnn** [comment]
Thanks for the report @rwstapleton0 !  I've put up a fix for the docs here: #15248 (it's slightly different from what you've suggested because the issue was that the docs went out of sync with an existing test in the codebase, linked below).

https://github.com/MystenLabs/sui/blob/19c31ab9f41beac36c3a5c3e6bd60268b29a5ef4/crates/sui-framework/packages/sui-framework/tests/clock_tests.move#L10

**Comment by rwstapleton0** [comment]
@stefan-mysten @amnn thanks guys, loving sui, keep up the great work :)
